### PR TITLE
Fix "Create snapshot" dialog validation

### PR DESCRIPTION
**Pull request by martinpitt**
https://issues.redhat.com/browse/RHEL-24794
https://issues.redhat.com/browse/RHEL-24793

### CODE REVIEW

File: src/components/vm/snapshots/vmSnapshotsCreateModal.jsx [hunk 1]
```diff
@@ -91,6 +91,7 @@ const MemoryPathRow = ({ onValueChanged, memoryPath, validationError }) => {
             <FileAutoComplete
                 onChange={value => onValueChanged("memoryPath", value)}
                 superuser="try"
+                isOptionCreatable
```
Comment (cockpituous) [inline src/components/vm/snapshots/vmSnapshotsCreateModal.jsx:94]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1435-20240209-085940-cf8791d4-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

Comment (cockpituous) [inline src/components/vm/snapshots/vmSnapshotsCreateModal.jsx:94]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1435-20240209-093602-b1c5a2e1-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

Comment (cockpituous) [inline src/components/vm/snapshots/vmSnapshotsCreateModal.jsx:94]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1435-20240209-100226-b1c5a2e1-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/components/vm/snapshots/vmSnapshotsCreateModal.jsx [hunk 2]
```diff
@@ -127,15 +127,15 @@ export class CreateSnapshotModal extends React.Component {
         this.setState({ dialogError: text, dialogErrorDetail: detail });
     }
 
-    onValidate(submitted = false) {
+    onValidate() {
```
Comment (jelly) [inline src/components/vm/snapshots/vmSnapshotsCreateModal.jsx:131]: Real minor nit, there is a typo in the commit message `should not bet empty`

Comment (jelly) [reply src/components/vm/snapshots/vmSnapshotsCreateModal.jsx:131]: Another minor nit and something we do super inconsistent in Cockpit is setting `isRequired` on the TextInput so the dialog shows a hint what is optional or not.

image

**Review by jelly — APPROVED** [review]
Thanks!

File: src/components/vm/snapshots/vmSnapshotsCreateModal.jsx (hunk 2 above)
Comment (martinpitt) [reply src/components/vm/snapshots/vmSnapshotsCreateModal.jsx:131]: We only use it in one place in cockpit (kdump dialog), once in podman (publish port), and not at all in c-machines. So perhaps right now it's best to not start with that in c-machines, but rather do that consistently on all dialogs, in a separate PR?

**Pull request merged**